portal-bridge/src/beacon_bridge.rs

```diff
@@ -0,0 +1,237 @@
+            }
+        });
+
+        // Serve `LightClientOptimisticUpdate` data
+        tokio::spawn(async move {
+            if let Err(err) = Self::serve_light_client_optimistic_update(api, portal_clients).await
+            {
+                warn!("Failed to serve light client optimistic update: {err}");
+            }
+        });
+    }
+
+    #[allow(dead_code)] // FIXME: Remove this once we use it
+    async fn serve_light_client_bootstrap(&self, block_root: String) -> anyhow::Result<()> {
+        let data = self.api.get_lc_bootstrap(block_root).await?;
+        let update: Value = serde_json::from_str(&data)?;
+        let _: LightClientBootstrapCapella = serde_json::from_value(update["data"].clone())?;
+        Ok(())
+    }
+
+    async fn serve_light_client_update(
+        api: ConsensusApi,
+        portal_clients: Vec<HttpClient>,
+        current_period: Arc<AtomicUsize>,
+    ) -> anyhow::Result<()> {
+        let expected_current_period = expected_current_slot(BEACON_GENESIS_TIME) / (32 * 256);
+        match (expected_current_period as usize)
+            .cmp(&current_period.load(std::sync::atomic::Ordering::Relaxed))
+        {
+            Ordering::Equal => {
+                // We already gossiped the latest data from the current period, no need to serve it again.
+                return Ok(());
+            }
+            Ordering::Less => {
+                // if we panic here, it is a bug
+                panic!("Expected current period is less than known period");
+            }
+            Ordering::Greater => {
+                // Continue
+            }
+        }
+
+        let data = api.get_lc_updates(expected_current_period, 1).await?;
+        let update: Value = serde_json::from_str(&data)?;
+        let update: LightClientUpdateCapella = serde_json::from_value(update[0]["data"].clone())?;
+        let fork_versioned_update = ForkVersionedLightClientUpdate {
+            fork_name: ForkName::Capella,
+            update: LightClientUpdate::Capella(update.clone()),
+        };
+
+        let content_value = BeaconContentValue::LightClientUpdatesByRange(
+            LightClientUpdatesByRange(VariableList::from(vec![fork_versioned_update])),
+        );
+        let content_key =
+            BeaconContentKey::LightClientUpdatesByRange(LightClientUpdatesByRangeKey {
+                start_period: expected_current_period,
+                count: 1,
+            });
+        info!(
+            "Got lc update for slot {:?}",
+            update.attested_header.beacon.slot
+        );
+
+        // Update the current known period if we successfully gossiped the latest data.
+        match Self::gossip_beacon_content(&portal_clients, content_key, content_value).await {
+            Ok(_) => {
+                current_period.store(
+                    expected_current_period as usize,
+                    std::sync::atomic::Ordering::Relaxed,
```

Comment:
I'm still figuring out the details of the atomic ordering semantics. So yeah, I'll admit some ignorance there. How sure are you that `Relaxed` reads and writes won't cause some kind of weird access order bug? I can't think of a specific problem, but I tend to just use the strictest-possible option, unless I take the time to prove to myself that it's definitely not a problem.

Reply:
Good point about more restrictiveness. Changed to `Aquire` and `Release` according to https://stackoverflow.com/questions/30407121/which-stdsyncatomicordering-to-use.